File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        vector<int> freq(1e5+1);
        int ans = 0;
        for(int i=0; i<nums.size(); i++){
            freq[nums[i]]++;
        }
        
        for(int i=0; i<=1e5; i++){
            if(freq[i]>1){
                if(i==1e5){
                    ans+=((freq[i]-1)*freq[i])/2;
                }else{
                    ans+=freq[i]-1;
                    freq[i+1]+=freq[i]-1;
                } 
            }
        }
        return ans;
    }
};

File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.ts
function minIncrementForUnique(nums: number[]): number {
    nums.sort((x,y)=>x-y);
    let ans:number = 0;
    for(let i=0; i<nums.length; i++){
        if(nums[i]<=nums[i-1]){
            ans += nums[i-1]-nums[i]+1;
            nums[i] = nums[i-1]+1;
        }
    }
    return ans;
};